src/main/java/gregicadditions/machines/TileEntityProcessingArray.java

```diff
@@ -417,35 +471,195 @@ protected boolean setupAndConsumeRecipeInputs(Recipe recipe) {
 				recipe.matches(true, importInventory, importFluids);
 		}
 
+		protected boolean didMachinesChange(ItemStack newMachineStack) {
+
+			if(newMachineStack == null || this.oldMachineStack == null) {
```

Comment:
If both stacks are null, this will return true and then go the `findRecipe` method, where one of the first checks is if the machine stack is null, which will return a null recipe, so I think that this is fine that the case for both machines stacks being null is not covered

Reply:
if we leave it like this, i would suggest at least renaming the method to something like `shouldRecheckRecipes`, since the current name would no longer be an accurate description. on the other hand, dealing with that other case here could save a few method calls

Reply:
You do have a good point that it doesn't quite do what the name implies. The logic as it is here is: "is at least one of the ItemStacks `null` or are the stacks not equal."

You have to be sure that neither stack is null before calling the equality check method or you'll get an NPE, but if both stacks are null the result of the method should probably be `false` since the stack didn't change since the last check.

Can probably just do:
```java
protected boolean didMachinesChange(ItemStack newMachineStack) {
    if(newMachineStack == null || this.oldMachineStack == null)
        return newMachineStack != this.oldMachineStack;

    return !ItemStack.areItemStacksEqual(this.oldMachineStack, newMachineStack);
}
```

Which would return `true` if precisely one of the two stacks are `null` (or neither are null and the stacks differ), or `false` if both stacks are `null`.